pom.xml

```diff
@@ -376,6 +376,36 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-pmd-plugin</artifactId>
+                <version>3.17.0</version>
+                <inherited>false</inherited>
+                <!-- aggregate configuration, parent build-->
+                <configuration>
+                    <printFailingErrors>true</printFailingErrors>
+                    <rulesets>
+                        <ruleset>${project.basedir}/pmd/legend-ruleset.xml</ruleset>
+                    </rulesets>
+                    <rules>
+                        <requireFilesExist>
+                            <files>
+                                <file>${project.basedir}/pmd/legend-ruleset.xml</file>
+                            </files>
+                        </requireFilesExist>
+                    </rules>
+                </configuration>
+                <executions>
+                    <execution>
+                        <id>pmd-check-aggregate</id>
+                        <phase>validate</phase>
+                        <goals>
+                            <goal>aggregate-pmd-check</goal>
+                        </goals>
+                    </execution>                    
+                </executions>
+            </plugin>
+
```

Comment:
I think we should flip this around and run from the children instead of the parent. I noticed that when running `aggregate-pmd-check` from the parent, it roughly doubles the time the build takes. But when running `check` from children, it adds only a small amount of time to the build. As far as I can tell, the same checks are done either way, and violations fail the build. Running from the parent means you get all the violations from the whole project at failure time, but I don't think it's worth doubling the build time for that.

```suggestion
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-pmd-plugin</artifactId>
                <version>3.17.0</version>
                <configuration>
                    <printFailingErrors>true</printFailingErrors>
                    <rulesets>
                        <ruleset>${project.parent.basedir}/pmd/legend-ruleset.xml</ruleset>
                    </rulesets>
                    <rules>
                        <requireFilesExist>
                            <files>
                                <file>${project.parent.basedir}/pmd/legend-ruleset.xml</file>
                            </files>
                        </requireFilesExist>
                    </rules>
                </configuration>
                <executions>
                    <execution>
                        <id>pmd-check</id>
                        <phase>validate</phase>
                        <goals>
                            <goal>check</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```